### PR TITLE
Remove argument transformers support

### LINKED ISSUE #122: Remove $argumentTransformers support
`$argumentTransformers` are defined as an array of functions that transform messages based on the index of the stream. The usage is awkward as we're moving away from index based references to stream.

We should remove the argument transformers for the node-stream and always provide the full message to the function. The function author may decompose messages from the stream as they desire.

request-reply functions should continue to operate on the message payload by default, and may opt-in to the whole message.

### PR DESCRIPTION
**Pull request by fbiville**
Fixes #122

### CODE REVIEW

**Comment by fbiville** [comment]
Note that this is open for discussion, not necessarily for merging yet.

This PR differs from #122 initial requirements.

One reason is that there is currently no easy way to distinguish request-reply functions from streaming ones when it comes to input unmarshalling (the request-reply function has already been promoted at that point). 

In the PR current form, `Message` is not exposed at all to the function input streams, only the payload, even for streaming functions (this is the least incompatible path).
The only special support for `Message` is in the output marshaller. If the function produces a message, the payload and headers will be properly propagated in the corresponding output signal fields.

My proposal would be to start exposing the full Message to streaming functions by default as soon as we solve #87 but not before.

File: lib/streaming-pipeline.js [hunk 1]
```diff
@@ -222,11 +194,10 @@ module.exports = class StreamingPipeline extends Writable {
         }
     }
 
-    _wireInputs(transformers) {
-        const inputCount = transformers.length;
+    _wireInputs(inputCount) {
         console.debug(`Wiring ${inputCount} input stream(s)`);
         for (let i = 0; i < inputCount; i++) {
-            const unmarshaller = new InputUnmarshaller(transformers[i]);
+            const unmarshaller = new InputUnmarshaller();
```
[the remark this answers is not in the document]
Comment (fbiville) [reply lib/streaming-pipeline.js]: This is not strictly necessary anymore, but I would rather keep it like this, given:

- this is symmetric with marshallers
- we may need again a specific unmarshaller per input stream for #87

**Comment by fbiville** [comment]
My only grief (I hope it's not too strong of a word for what I want to mean) is that exposing Messages as the only option for streaming functions dilutes the business code in technical details for simple streaming cases. On the other side, I recognize this will help with more complicated scenarios (that require access to header values etc etc).

I think the side effect of #87 will be that users can get some control back and expose only what's necessary to them so we're just in a temporary situation, anyway.

**Comment by fbiville** [comment]
PR (painfully) rebased by the way.

File: lib/streaming-pipeline.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (fbiville) [reply lib/streaming-pipeline.js]: Also, I reinstated a guard in the input unmarshaller just to make sure a given unmarshaller instance deals with the right input stream (better safe than sorry).

File: lib/input-unmarshaller.js [hunk 2]
```diff
@@ -8,21 +8,24 @@ const {
...
-    const messageBuilder = Message.builder();
-    for (const headerName in headers) {
-        messageBuilder.addHeader(headerName, headers[headerName]);
-    }
-    return messageBuilder.payload(payload).build();
-};
-
 module.exports = class InputUnmarshaller extends Transform {
-    constructor(argumentTransformer) {
+    constructor(expectedInputIndex) {
```
Comment (scothis) [inline lib/input-unmarshaller.js:12]: why do we care about the input index? I don't see anything that varies depending on the input stream? We could make InputUnmarshaller a singleton

**Pull request merged**